### PR TITLE
[OPIK-4335] [SDK] Add G-Eval metric to TypeScript SDK

### PR DESCRIPTION
**Pull request by petrotiurin**
## Details

This PR adds the GEval metric implementation to the TypeScript SDK, mirroring the functionality already available in the Python SDK. GEval is a task-agnostic LLM-as-a-judge metric that allows users to specify custom task descriptions and evaluation criteria.

**Key Features:**
- **Core GEval Implementation**: Base `GEval` class with custom `taskIntroduction` and `evaluationCriteria` parameters
- **Chain of Thought (CoT) Caching**: LRU cache for CoT prompts to avoid redundant LLM calls
- **Logprobs Support**: Uses logprobs when available (via Vercel AI SDK `providerOptions`) for weighted average scoring, falling back to string parsing
- **13 Built-in Judge Classes**: Pre-configured judges for common evaluation scenarios:
  - `QARelevanceJudge` - Answer relevance to questions
  - `SummarizationConsistencyJudge` - Summary faithfulness
  - `SummarizationCoherenceJudge` - Summary structure and clarity
  - `DialogueHelpfulnessJudge` - Assistant helpfulness in dialogue
  - `DemographicBiasJudge`, `GenderBiasJudge`, `PoliticalBiasJudge`, `ReligiousBiasJudge`, `RegionalBiasJudge` - Bias detection
  - `AgentTaskCompletionJudge` - Agent task fulfillment
  - `AgentToolCorrectnessJudge` - Agent tool usage correctness
  - `PromptUncertaintyJudge` - Prompt ambiguity detection
  - `ComplianceRiskJudge` - Regulatory compliance risk assessment
- **GEvalPreset Class**: Allows instantiation using preset names (backward compatibility)
- **Comprehensive Test Coverage**: Unit tests for parser, template generation, and all judge classes, plus integration tests

**Implementation Details:**
- Follows TypeScript SDK patterns: extends `BaseLLMJudgeMetric`, uses Zod for validation
- Score normalization: Judges return 0-10 integers, normalized to 0.0-1.0
- Model configuration: Supports model IDs, LanguageModel instances, and OpikBaseModel instances
- Generation parameters: Supports `temperature`, `seed`, `maxTokens`, and `modelSettings`

## Change checklist
- [x] User facing
- [x] Documentation update

## Issues

- OPIK-4335

## Testing

**Unit Tests:**
- Parser tests: String parsing, logprobs parsing, edge cases (null scores, out-of-range scores, wrong keys)
- Template tests: CoT and query prompt generation
- GEval tests: CoT caching, scoring logic, preset instantiation, model settings propagation
- Judge class tests: All 13 judges instantiation, naming, preset mapping, inheritance, constructor options

**Integration Tests:**
- Custom GEval instance with custom task/criteria
- QARelevanceJudge preset usage
- End-to-end scoring with real LLM calls

**Test Coverage:**
- 95 unit tests passing
- 2 integration tests added
- All edge cases covered (null handling, boundary scores, logprobs edge cases)

## Documentation

**Updated Files:**
- `apps/opik-documentation/documentation/fern/docs/evaluation/metrics/g_eval.mdx`
  - Added TypeScript code examples alongside Python examples for all GEval judges
  - Updated "Customising models" section with TypeScript examples
  - All examples now use CodeBlocks component with side-by-side Python/TypeScript tabs

- `apps/opik-documentation/documentation/fern/docs/reference/typescript-sdk/evaluation/metrics.mdx`
  - Added comprehensive "GEval" section covering:
    - Core GEval metric usage and parameters
    - How GEval works (CoT generation + scoring)
    - All 13 built-in judge classes with examples
    - Score interpretation and configuration options

**Documentation Features:**
- Complete TypeScript examples for all built-in judges
- Parameter documentation for GEval and all judge classes
- Usage examples demonstrating real-world scenarios
- Integration with existing LLM Judge Metrics documentation structure

### CODE REVIEW

File: sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/parser.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    const entry = entries[SCORE_TOKEN_POSITION];
+    const topLogprobs = entry.top_logprobs ?? [];
+    const tokenCandidate = String(entry.token ?? "");
+
+    let linearProbsSum = 0.0;
+    let weightedScoreSum = 0.0;
+
+    for (const candidate of topLogprobs) {
+      const tokenStr = String(candidate.token ?? "");
+      if (!/^\d+$/.test(tokenStr)) continue;
```
[the remark this answers is not in the document]
Comment (petrotiurin) [reply sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/parser.ts:117]: This shouldn't be an issue as we're parsing a valid structured output. If output is not tokenised correctly then there might be an issue with the output structure altogether and the error message is appropriate like we do right now.

File: sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/GEval.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+    const modelOptions = this.buildModelOptions();
+    const generated = await this.model.generateString(
+      prompt,
+      undefined,
+      modelOptions
+    );
+
+    storeCoT(key, generated);
+    return generated;
```
[the remark this answers is not in the document]
Comment (petrotiurin) [reply sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/GEval.ts:137]: Filed OPIK-4352 to address this together with Python SDK.

File: sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/parser.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  if (!providerMetadata) return undefined;
+
+  for (const providerKey of Object.keys(providerMetadata)) {
+    const providerData = providerMetadata[providerKey] as
+      | Record<string, unknown>
+      | undefined;
+    if (providerData && providerData.logprobs) {
+      return providerData.logprobs as LogprobEntry[][];
+    }
+  }
```
[the remark this answers is not in the document]
Comment (petrotiurin) [reply sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/parser.ts:73]: no it doesn't, but I've simplified this part to use types from vercel ai

**Comment by petrotiurin** [comment]
E2E test issues are due to the openai API limit we're hitting company-wide. Will rerun once that's resolved.

File: sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/GEval.ts [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    try {
+      const providerResponse = await this.model.generateProviderResponse(
+        [{ role: "user", content: llmQuery }],
+        {
+          ...modelOptions,
+          output: Output.object({
+            schema: responseSchema,
+          }),
+          providerOptions: {
+            openai: { logprobs: true, top_logprobs: 20 },
```
Comment (vincentkoc) [inline sdks/typescript/src/opik/evaluation/metrics/llmJudges/gEval/GEval.ts]: @petrotiurin are model params provider specific, if so how do we map these?

**Review by awkoy — APPROVED** [review]
Amazing work 👍
